### PR TITLE
New Admin Page: My Jetpack (connection management)

### PR DESCRIPTION
**Pull request by dereksmart**
Note: this PR does not include the dotcom changes that will be made on master user switch.  That will be done in a following PR.

On this page, users will be able to view their connection information, as well as the ability to transfer "site ownership" aka master user.  

You will need to run `grunt` locally to see the css file changes.  

Prior to testing, please disconnect jetpack entirely. 

Begin testing:
Set up 3 separate user accounts; (at least) Two admins and one editor (or any non-admin).  I found it's easiest to log in to all three accounts in different windows/browsers.  

As the first admin: 
- Connect Jetpack to your wpcom account.  This will assign you as the master user. 

As the second admin: 
- Log in, click on "My Jetpack": You should be able to see the primary connection holder, and a CTA to link the site to your wordpress.com account
- Link your account to a unique wpcom account from the master.  
- Look at "My Jetpack" again. You should now be able to switch the primary account to your connection.  You can also unlink your account.  

As the non-admin: 
Repeat steps from the second admin.  The Only difference should be that you can't do/see anything but link/unlink your own wpcom account to the site.  

Some notes about My Jetpack
- Only admins of the site will be able to disconnect the Jetpack site entirely. 
- You may now switch Master user roles
- Only able to assign users as master who are both Admins _and_ linked to a dotcom account. 
- You should only see yourself and the master user's info at any given time.  If you are the master user, you will only see yourself.  
- Users below admin level still have access to the page, however they can only see their own connection status.  
- You should only be able to see the page if the site is connected.

### CODE REVIEW

File: views/admin/my-jetpack-page.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	<?php Jetpack::init()->load_view( 'admin/network-activated-notice.php' ); ?>
+
+	<?php do_action( 'jetpack_notices' ) ?>
+
+	<div id="my-jetpack-page-template"></div>
+
+	<script id="tmpl-connection-page" type="text/html">
+		<div class="content-container">
+			<div id="my-jetpack-content" class="content">
+				<h2><?php _e( 'Jetpack Connection Status' ); ?></h2>
```
Comment (jeherve) [inline views/admin/my-jetpack-page.php]: Seems like there is a missing jetpack textdomain here.

File: views/admin/my-jetpack-page.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+							</div>

+						</div>
+
+						<?php // right col ( Link/Unlink my account ) ?>
+						<div class="j-col j-lrg-4 j-md-12 j-sm-12 wp-action">
+							<h3 title="<?php _e( 'Account Actions', 'jetpack' ); ?>"><?php _e( 'Account Actions', 'jetpack' ); ?></h3>
+							<div class="action-btns">
+								<# if ( data.currentUser.isUserConnected ) { #>
+									<a class="button" title="Disconnect your WordPress.com account from Jetpack" href="<?php echo wp_nonce_url( Jetpack::admin_url( 'action=unlink' ), 'jetpack-unlink' ); ?>"><?php esc_html_e( 'Unlink my account ', 'jetpack' ); ?></a>
```
Comment (jeherve) [inline views/admin/my-jetpack-page.php]: The link title should be translatable as well.

File: views/admin/my-jetpack-page.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+								<span>{{{ data.masterUser.masterDataCom.login }}}</span>
+							</div>

+						</div>
+
+						<?php // right col ( Change primary user ) ?>
+						<div class="j-col j-lrg-4 j-md-12 j-sm-12 wp-action">
+							<h3 title="<?php _e( 'Account Actions', 'jetpack' ); ?>"><?php _e( 'Change Primary User', 'jetpack' ); ?></h3>
+							<div class="action-btns">
+								<a class="button" title="Change the primary account holder" id="change-primary-btn"><?php esc_html_e( 'Change Primary', 'jetpack' ); ?></a>
```
Comment (jeherve) [inline views/admin/my-jetpack-page.php]: The link title can't be translated.

File: views/admin/my-jetpack-page.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+										$all_users = get_users();
+										foreach ( $all_users as $user ) {
+											if ( $user->ID != Jetpack_Options::get_option( 'master_user' ) && Jetpack::is_user_connected( $user->ID ) && $user->caps['administrator'] ) {
+												echo "<option value='{$user->ID}'>$user->display_name</option>";
+											}
+										}
+										?>
+									</select>
+									<?php wp_nonce_field( 'jetpack_change_primary_user', '_my_jetpack_nonce' ); ?>
+									<input type="submit" name="jetpack-set-master-user" id="save-primary-btn" class="button button-primary" value="Save" title="Set the primary account holder"/>
```
Comment (jeherve) [inline views/admin/my-jetpack-page.php]: The input field's title should be translatable.

File: views/admin/my-jetpack-page.php [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+			<?php // Disconnect Site Button ?>
+			<?php if ( current_user_can( 'jetpack_configure_modules' ) ) : ?>
+				<div class="j-row disconnect">
+					<div class="j-col j-lrg-12 j-md-12 j-sm-12">
+						<a class="button" id="jetpack-disconnect" href="#"><?php esc_html_e( 'Disconnect site from WordPress.com', 'jetpack' ); ?></a>
+					</div>
+				</div>
+
+				<div id="jetpack-disconnect-content">
+					<h2>Disconnecting Jetpack</h2>
```
Comment (jeherve) [inline views/admin/my-jetpack-page.php]: This should be translated as well.

File: _inc/jp-connection.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        };
+
+    $( document ).ready(function () {
+        renderPageTemplate( data );
+
+        // Set someone as master.
+        $( '#change-primary-btn' ).click( function() {
+
+            if ( false === data.otherAdminsLinked ) {
+                window.alert( 'You must link another admin account before switching primary account holders.' );
```
Comment (jeherve) [inline _inc/jp-connection.js]: This will need to be translated as well.

File: _inc/header.php [hunk 7]
```diff
@@ -6,16 +6,21 @@
...
 					<li class="jetpack-modules">
 						<a href="<?php echo Jetpack::admin_url( 'page=jetpack_modules' ); ?>" class="jp-button--settings <?php if ( 'jetpack_modules' == $current ) { echo 'current'; } ?>"><?php esc_html_e( 'Settings', 'jetpack' ); ?></a>
 					</li>
 					<li class="jetpack-modules">
 						<a href="http://jetpack.me/survey/?rel=<?php echo JETPACK__VERSION; ?>" class="jp-button--settings"><?php esc_html_e( 'Feedback', 'jetpack' ); ?></a>
 					</li>
-					<?php endif; ?>
+						<?php endif; // End if admin ?>
+					<li class="jetpack-modules">
+						<a title="View your Jetpack connection details" href="<?php echo Jetpack::admin_url( 'page=my_jetpack' ); ?>" id="my-jetpack" class="jp-button--settings"><?php esc_html_e( 'My Jetpack', 'jetpack' ); ?></a>
```
Comment (kraftbj) [inline _inc/header.php]: Should we translate the `title` attribute?

File: views/admin/my-jetpack-page.php [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+					<div class="j-col j-lrg-12 j-md-12 j-sm-12">
+						<a class="button" id="jetpack-disconnect" href="#"><?php esc_html_e( 'Disconnect site from WordPress.com', 'jetpack' ); ?></a>
+					</div>
+				</div>
+
+				<div id="jetpack-disconnect-content">
+					<h2><?php _e( 'Disconnecting Jetpack', 'jetpack' ); ?></h2>
+					<p><?php _e( 'Before you completely disconnect Jetpack is there anything we can do to help?', 'jetpack' ); ?></p>
+					<a class="button" title="<?php esc_attr_e( 'Disconnect Jetpack', 'jetpack' ); ?>" href="<?php echo wp_nonce_url( Jetpack::admin_url( 'action=disconnect' ), 'jetpack-disconnect' ); ?>">Confirm Disconnect</a>
+					<a class="button primary" target="_blank" title="<?php esc_attr_e( 'Jetpack Support', 'jetpack' ); ?>" href="http://jetpack.me/contact-support/">I Need Support</a>
```
Comment (kraftbj) [inline views/admin/my-jetpack-page.php]: Need to wrap `I need support` to be translatable.

**Comment by dereksmart** [comment]
closing in favor of #2268 - this was has been gaffed by rebase

**Pull request closed without merging**